src/content/reducers/url-state.js

```diff
@@ -151,6 +166,44 @@ function hidePlatformDetails(state: boolean = false, action: Action) {
   }
 }
 
+function threadOrder(state: ThreadIndex[] = [], action: Action) {
+  switch (action.type) {
+    case 'RECEIVE_PROFILE_FROM_ADDON':
+    case 'RECEIVE_PROFILE_FROM_WEB':
+    case 'RECEIVE_PROFILE_FROM_FILE': {
+      // When receiving a new profile, try to use the thread order specified in the URL,
+      // but ensure that the IDs are correct.
+      const threads = defaultThreadOrder(action.profile.threads);
+      const validURLThreads = state.filter(index => threads.includes(index));
+      const missingThreads = threads.filter(index => !state.includes(index));
+      return validURLThreads.concat(missingThreads);
+    }
+    case 'CHANGE_THREAD_ORDER': {
+      const { threadOrder } = action;
+      return threadOrder;
+    }
+    default:
+      return state;
+  }
+}
+
+function hiddenThreads(state: ThreadIndex[] = [], action: Action) {
+  switch (action.type) {
+    case 'HIDE_THREAD': {
+      const { threadIndex, threadOrder } = action;
+      // Do not allow hiding the last thread.
+      const isLastVisibleThread = state.length + 1 === threadOrder.length;
```

Comment:
Maybe that's just me, but I'd write the comparison like this; `threadOrder.length - state.length <= 1`